src/core/main.c

```diff
@@ -1598,11 +1598,19 @@ static void initialize_clock(void) {
                  */
                 (void) clock_reset_timewarp();
 
-        r = clock_apply_epoch();
-        if (r < 0)
-                log_error_errno(r, "Current system time is before build time, but cannot correct: %m");
-        else if (r > 0)
-                log_info("System time before build time, advancing clock.");
+        ClockChangeDirection change_dir;
+        r = clock_apply_epoch(&change_dir);
+        if (r < 0) {
+                if (change_dir == CLOCK_CHANGE_FORWARD)
+                        log_error_errno(r, "Current system time is before build time, but cannot correct: %m");
+                else if (change_dir == CLOCK_CHANGE_BACKWARD)
+                        log_error_errno(r, "Current system time is far ahead build time, but cannot correct: %m");
+        } else if (r > 0) {
+                if (change_dir == CLOCK_CHANGE_FORWARD)
+                        log_info("System time before build time, advancing clock.");
+                else if (change_dir == CLOCK_CHANGE_BACKWARD)
+                        log_info("System time far ahead build time, reversing clock.");
```

Comment:
I think proper english would be "far ahead of".

maybe even add a "too" in there. But maybe mention the extra time we give here.

```
log_info("System time further ahead than %s after build time, resetting clock to build time.", FORMAT_TIMESPAN(CLOCK_VALID_RANGE_USEC_MAX, USEC_PER_DAY));
```